packages/widget/src/controllers/transfers/evm/build.ts

```diff
@@ -46,9 +46,15 @@ export async function buildEvmFungibleTransactions(
     this.resourceAmount.toString()
   );
   const originalFee = await evmTransfer.getFee(originalTransfer);
+  // NOTE: for percentage fee, if both are equal, it means we can calculate the amount with fee avoiding second subtraction
+  const calculateAmountWithFee =
+    originalFee.type === FeeHandlerType.PERCENTAGE &&
+    this.resourceAmount.toString() ===
```

Comment:
is this check needed?
in what case will the `resourceAmount` be different from `originalTransfer.details.amount`

Reply:
the issue was related to entering the number and resourceAmount being subtracted twice. So if you enter 12, although `originalTransfer` object would have 12 as amount, `resourceAmount` would be first 12, then 11 and then 10, which will be incorrectly displayed. With this check, if you enter any amount and you are using `percentage` as fee, it will display the correct amount to receive